service/client_user_test.go

```diff
@@ -169,24 +170,137 @@ func Test_authenticateUser(t *testing.T) {
 	}
 }
 
+func Test_verifyAuthToken(t *testing.T) {
+	validEmail := randomdata.Email()
+	validPassword := "Abcd!123@"
+	resp, err := userSvc.createUser(
+		&pblib.User{
+			FirstName:    randomdata.FirstName(randomdata.Male),
+			LastName:     randomdata.LastName(),
+			Email:        validEmail,
+			Password:     validPassword,
+			Organization: testOrg,
+		})
+	assert.Nil(t, err, "Test_verifyAuthToken")
+	assert.NotNil(t, resp, "Test_verifyAuthToken")
+	emailToken := resp.GetIdentification().GetToken()
+	err = userSvc.verifyEmailToken(emailToken)
+	assert.Nil(t, err, "verify the email")
+
+	resp, err = userSvc.authenticateUser(validEmail, validPassword)
+	assert.Nil(t, err, "Test_verifyAuthToken")
+	assert.NotNil(t, resp, "Test_verifyAuthToken")
+	authToken := resp.GetIdentification().GetToken()
+	authSecret := resp.GetIdentification().GetSecret()
+
+	cases := []struct {
+		desc     string
+		token    string
+		isExpErr bool
+		errStr   string
+	}{
+		{
+			"test for valid auth token",
```

Comment:
where do you test for expiration?

Reply:
I can't unless we can somehow insert an expired auth token in the DB.
It is currently implemented like https://github.com/hwsc-org/hwsc-user-svc/blob/master/service/service.go#L589